pkg/snapshot/request.go

```diff
@@ -227,3 +231,84 @@ func CreateSnapshotOptionsSecret(vClusterNamespace, vClusterName string, snapsho
+	// First, try to get saved snapshots
+	restoreClient := RestoreClient{
+		Snapshot: *snapshotOpts,
+	}
+
+	var snapshotRequests []Request
+	savedSnapshotRequest, err := restoreClient.GetSnapshotRequest(ctx)
+	if err != nil {
+		log.Debugf("failed to get saved snapshot request: %v", err)
+	}
+	if savedSnapshotRequest != nil {
+		// The snapshot request has been saved while it was in progress (it's
+		// set to Completed/PartiallyFailed after the upload). Therefore, here
+		// we update the phase to the correct final state.
+		if savedSnapshotRequest.Spec.IncludeVolumes {
+			if savedSnapshotRequest.Status.VolumeSnapshots.Phase == volumes.RequestPhaseCompleted {
+				savedSnapshotRequest.Status.Phase = RequestPhaseCompleted
+			} else {
+				savedSnapshotRequest.Status.Phase = RequestPhasePartiallyFailed
+			}
+		} else {
+			savedSnapshotRequest.Status.Phase = RequestPhaseCompleted
+		}
+		snapshotRequests = append(snapshotRequests, *savedSnapshotRequest)
+	}
+
+	listRequests := true
+	var continueOption string
+	for listRequests {
+		listOptions := metav1.ListOptions{
+			LabelSelector: constants.SnapshotRequestLabel,
+			Continue:      continueOption,
+		}
+		snapshotRequestConfigMaps, err := kubeClient.CoreV1().ConfigMaps(vClusterNamespace).List(ctx, listOptions)
+		if err != nil {
+			return fmt.Errorf("failed to list snapshot request ConfigMaps: %w", err)
+		}
+		for _, configMap := range snapshotRequestConfigMaps.Items {
+			snapshotRequest, err := UnmarshalSnapshotRequest(&configMap)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal snapshot request from ConfigMap %s/%s: %w", configMap.Namespace, configMap.Name, err)
+			}
+			if snapshotRequest.Spec.URL != snapshotOpts.GetURL() {
+				continue
+			}
+			if savedSnapshotRequest != nil &&
+				(snapshotRequest.Name == savedSnapshotRequest.Name ||
+					snapshotRequest.Spec.URL == savedSnapshotRequest.Spec.URL &&
+						snapshotRequest.CreationTimestamp.Time.Before(savedSnapshotRequest.CreationTimestamp.Time)) {
+				// Skip the local snapshot request if:
+				// 1. it's the same request as the uploaded one, or
+				// 2. it's older than the saved one.
+				continue
+			}
+
+			snapshotRequests = append(snapshotRequests, *snapshotRequest)
+		}
+
+		continueOption = snapshotRequestConfigMaps.Continue
+		listRequests = snapshotRequestConfigMaps.Continue != ""
+	}
+
+	if len(snapshotRequests) == 0 {
+		log.Errorf("vCluster snapshot %q not found", snapshotOpts.GetURL())
+		return nil
+	}
+
+	header := []string{"SNAPSHOT", "STATUS", "AGE"}
+	values := make([][]string, len(snapshotRequests))
+	for i, snapshotRequest := range snapshotRequests {
+		values[i] = []string{
+			snapshotRequest.Spec.URL,
+			string(snapshotRequest.Status.Phase),
+			duration.HumanDuration(time.Since(snapshotRequest.CreationTimestamp.Time)),
+		}
+	}
+	table.PrintTable(log, header, values)
```

Comment:
I feel like this table should be generated on the [GetSnapshots](https://github.com/loft-sh/vcluster/pull/3269/files/ad7ebe77d6dd6297e5a8bb94b92a6972e17e7b93..197e8fa1e9111d04266f88027437c3082555cd94#diff-caa51c5e4491f90f012a7676cec7a7562e752a17899f4b3cf90be802443769a0R66) func in the cli package and this function should return a list of snapshot request `snapshotRequests []Request`

Reply:
Yeah, decoupling is not good here, printing should definitely be moved out, maybe not into the CLI package, because vcluster binary is using this as well. I'll see how to change this.